src/coreclr/src/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -330,6 +331,68 @@ private PdbSymbolReader OpenAssociatedSymbolFile(string peFilePath, PEReader peR
 
             return reader;
         }
+
+        private object _normalizedLock = new object();
+        private HashSet<TypeDesc> _nonNormalizedTypes = new HashSet<TypeDesc>();
+        private Dictionary<TypeDesc, TypeFlags> _normalizedTypeCategory = new Dictionary<TypeDesc, TypeFlags>();
+
+        public TypeFlags NormalizedCategoryFor4ByteStructOnX86(TypeDesc type)
```

Comment:
There's a discussion about adding x86 support to Native AOT - one of the the things we would want to do is make RyuJIT changes so that we don't have to compensate for 20 year old legacy quirks in x86 RyuJIT. Do you know if this is one of such quirks? If so, I would prefer we put it in a R2R-specific file.